blog/tests/views_test.py

```diff
@@ -14,116 +17,175 @@ class PostListTest(TestCase):
+        )
+        self.test_user2 = User.objects.create_user(
+            username='testuser2', password='12345'
         )
 
-        john = User.objects.create(username="john")
+        self.post_first = Post.objects.create(
+            author=self.user,
+            title='Title post first',
+            text='Text post first',
+            published_date=datetime(
+                day=8, month=11, year=2018, tzinfo=self.time_z
+            ),
+        )
 
         self.post_second = Post.objects.create(
-            author=john, title="Title John", text="Text John"
+            author=self.user,
+            title='Title post second',
+            text='Text post second',
+            published_date=datetime(
+                day=8, month=11, year=2016, tzinfo=self.time_z
+            ),
         )
+
         self.post_third = Post.objects.create(
-            author=john, title="Title John 2", text="Text John 2"
+            author=self.user,
+            title='Title post third',
+            text='Text post third',
+            published_date=datetime(
+                day=8, month=11, year=2014, tzinfo=self.time_z
+            ),
         )
-        self.post_third.publish()
 
     def tearDown(self):
         """Clean-up test data."""
+        del self.time_z
+        del self.client
+        del self.user
+        del self.test_user2
+
         del self.post_first
         del self.post_second
         del self.post_third
 
-    def test_post_list_correct_order(self):
-        """Test show post list on the page and test correct order."""
-        response = self.client.get(reverse("post_list"))
-        assert response.status_code == 200
-
-        self.assertTemplateUsed(response, "blog/post_list.html", "blog/base.html")
-
-        assert len(response.context["posts"]) == 2
-
-        first_title_on_the_page = response.context["posts"][0].title
-        first_text_on_the_page = response.context["posts"][0].text
-
-        second_title_on_the_page = response.context["posts"][1].title
-        second_text_on_the_page = response.context["posts"][1].text
+    def test_logged_correct_temp(self):
+        """The user is logged in and used the correct template."""
+        self.client.login(username='bob', password='password')
+        resp = self.client.get(reverse('post_list'))
 
-        assert first_title_on_the_page == "Title John 2"
-        assert first_text_on_the_page == "Text John 2"
+        self.assertEqual(str(resp.context['user']), 'bob')
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, 'blog/post_list.html')
 
-        assert second_title_on_the_page == "Title Bob"
-        assert second_text_on_the_page == "Text bob"
+    def test_post_list_correct_order(self):
+        """Test show post list on the page and test correct order."""
+        with patch(
+            'django.utils.timezone.now',
+            lambda: datetime(day=1, month=1, year=2018, tzinfo=self.time_z),
```

Comment:
You need to test 3 cases:
- the date is frozen before anything got posted
- the date is frozen after everything got posted
- the date is frozen in between of those